### ISSUE TITLE
Support for disabled ActionView's monkey patch 

### ISSUE THREAD

**Issue by kinduff** [jonspalmer/view_component-storybook#73]
Hello :wave: 

I was excited to use this library, sadly I'm implementing in a Rails < 6.0.1 application.

As stated in the docs, we need to disable the monkey patch: https://viewcomponent.org/compatibility.html#disabling-the-render-monkey-patch-rails--61

I'm looking for a way to modify the superclass `StoryConfig` but I've been unable to nail this one, since changing the `show.html.erb` to:

```erb
<%= render_component(@story.component.new(@story.constructor_args.kwargs)) do |c| -%>
  <% @story.slots.each do |slot| -%>
    <%- slot.call do -%>
      <%= instance_eval(&slot.content_block) if slot.content_block -%>
    <%- end -%>
  <%- end -%>
  <%- instance_eval(&@story.content_block) if @story.content_block -%>
<%- end -%>
```

Is of course not working at all.

Neither is:

```erb
<%= render_component(@story.component) %>
```

Since it raises an exception regarding a missing method in my component `render_in`.

Any pointers? Or support to these Rails versions is not in the roadmap?

**Comment by jonspalmer** [comment]
If you have the monkey patch enabled then the existing code should work. We have the test suite running against Rails 5.2 and 6.0 and things pass. You only need `render_component` if you **can't** use the monkey patch. It's possible we could support a monkey-patch disabled version but I haven't though how we'd do that. Happy to consider a PR.

**Comment by kinduff** [comment]
@jonspalmer Thanks for the quick reploy. I'm sorry, big typo from my side. We do have the monkey patch disabled.

Any pointers on how I could approach this? I gonna give it a stab this weekend.

**Comment by jonspalmer** [comment]
It looks like ViewComponent deals with exactly the same issue in Previews: https://github.com/github/view_component/blob/main/app/views/view_components/preview.html.erb

We a slightly more complicated view that previews so the exact pattern might not work. I could imagine two solutions:
1. a view helper that we invoke in `show.html.erb` that calls `send(:render)` or `send(:render_component)` based on the right logic or 2. We just duplicate `show.html.erb` and pick the view in the controller method. There might be other options. I think I prefer the former as its slightly more maintainable. I suspect it till need a bit of experimentation. 

Not the prior art for testing in the ViewComponent repo: https://github.com/github/view_component/blob/main/test/test_helper.rb#L112

We could add something similar but with an rspec shared_content that leverages an `around` block.

**Comment by kinduff** [comment]
Yes, that seems about right.

What about the `Story` and `StoryConfig` class? I see you are passing these to the `render` method, but it's puzzling me how it acts as a component, and how the arguments are being used.

Going to setup a demo application to tinker and develop this, thanks for the tips.

**Comment by jonspalmer** [comment]
I have a demo app here: https://github.com/jonspalmer/view_component_storybook_example might be a good starting point though you'd likely have to downgrade Rails. The only place I think we're calling render is in `show`. Which other cases are you referring to?

**Comment by jonspalmer** [comment]
Circling back to this conversation I'm curious if you have thoughts on the discussion in #150

**Comment by jonspalmer** [comment]
fixed in v1.0.0
